setup.py

```diff
@@ -134,19 +142,34 @@ def vitables_data_files():
 # ========== #
 # main stuff #
 # ========== #
-
-setup(name="liam2",
-      # cx_freeze wants only ints and dots
-      version=int_version('0.10.2'),
-      description="LIAM2",
-      cmdclass={"build_ext": MyBuildExt},
-      ext_modules=ext_modules,
-      options={"build_ext": build_ext_options, "build_exe": build_exe_options},
-      executables=[Executable("main.py")],
-      install_requires=['numpy', 'numexpr', 'tables'],
-      extras_require=dict(
-          interpolation=['bcolz'],
-          plot=['matplotlib'],
-          view=['vitables'],
-          ),
-      )
+if platform.system() != 'Linux':
+    setup(name="liam2",
+          # cx_freeze wants only ints and dots
+          version=int_version('0.10.2'),
+          description="LIAM2",
+          cmdclass={"build_ext": MyBuildExt},
+          ext_modules=ext_modules,
+          options={"build_ext": build_ext_options, "build_exe": build_exe_options},
+          executables=[Executable("main.py")],
+          install_requires=['numpy', 'numexpr', 'tables'],
+          extras_require=dict(
+              interpolation=['bcolz'],
+              plot=['matplotlib'],
+              view=['vitables'],
+              ),
+          )
+else:
+    setup(name="liam2",

```

Comment:
Ouch, this is ugly. Why not have something like an "other_kwargs" dict which would store extra arguments and would contain {'executables': [Executable("main.py")]} on non-linux platforms and then get rid of the if block? I also doubt this is the right way to go, see my general remarks.